### PR TITLE
[VALIDATION] updating validation and adding missing exception/schema …

### PR DESCRIPTION
**Pull request by jeffctr**
…files

### CODE REVIEW

File: src/Exception/Schema/DataEncode/DataEncodeException.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+namespace Shale\Exception\Schema\DataEncode;
```
Comment (andrewstudionone) [inline src/Exception/Schema/DataEncode/DataEncodeException.php:3]: Missing `declare(strict_types=1);` on all .php files

File: src/Exception/Schema/DataEncode/DataEncodeException.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+use Shale\Exception\Schema\SchemaException;
+
+class DataEncodeException extends SchemaException
+{
+    public $problemValue;
+
+    public function __construct(
+        string $message = "",
+        $problemValue = null,
+        $code = 0,
```
Comment (andrewstudionone) [inline src/Exception/Schema/DataEncode/DataEncodeException.php]: should it be `int $code = 0`?

File: src/Exception/Schema/DataEncode/DataEncodeException.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use Throwable;
+use Shale\Exception\Schema\SchemaException;
+
+class DataEncodeException extends SchemaException
+{
+    public $problemValue;
+
+    public function __construct(
+        string $message = "",
+        $problemValue = null,
```
Comment (andrewstudionone) [inline src/Exception/Schema/DataEncode/DataEncodeException.php:14]: Do you have specific type for `$problemValue` ?

**Review by andrewstudionone — APPROVED** [review]

File: src/Exception/Schema/DataEncode/DataEncodeException.php (hunk 3 above)
Comment (jeffctr) [reply src/Exception/Schema/DataEncode/DataEncodeException.php:14]: No

**Review by andrewstudionone — APPROVED** [review]

**Pull request merged**